### PR TITLE
Make existing api call function reusable for Rundeck

### LINKED ISSUE #16: Create reusable Rundeck api call utility
Subsequent tasks will require making Rundeck api calls. 

In this current issue, we want to accomplish the following:

That we have a generic Rundeck api call utility

### PR DESCRIPTION
**Pull request by kosyfrances**
Fixes #16

### CODE REVIEW

File: lib/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// MakeRequest makes an API request.
+// It returns a response object and an error object.
+func MakeRequest(payload interface{}, URL, method string) (*http.Response, error) {
```
Comment (osule) [inline lib/utils/utils.go]: A small nitpick here:
I'ld order the arguments as `method`, `URL`, `payload`.

File: lib/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	// Build the request
+	b, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("cannot marshal payload. error: %v", err)
+	}
+
+	body := bytes.NewReader(b)
+	req, err := http.NewRequest(method, URL, body)
+	if err != nil {
+		return nil, fmt.Errorf("cannot make HTTP request. error: %v", err)
```
Comment (osule) [inline lib/utils/utils.go]: `cannot create HTTP request` seems to be a better wording for the error message.
`make` can be jumbled with performing a request.

File: lib/utils/utils.go (hunk 1 above)
Comment (kosyfrances) [reply lib/utils/utils.go]: sounds good to me.

**Comment by kosyfrances** [comment]
Addressed the comments. Merging this. Thanks for the review :)

**Pull request merged**